torch/fx/experimental/symbolic_shapes.py

```diff
@@ -4017,6 +4017,7 @@ def add_expr(expr):
         add_expr(e)
         # Other relational expressions this expression implies
         if isinstance(e, sympy.Eq):
+            add_expr(sympy.Eq(e.rhs, e.lhs))
```

Comment:
btw, also noticed that the dualizing above didn't have an effect, I think canonicalizing after dualizing cancels out. maybe something to follow up in the future

Reply:
for the code above, we get something like `Eq(u0 - u1 - u2, 0)` to evaluate statically with substitutions like `Eq(u1 + u2 - u0, 0)`, which didn't pattern match.

Reply:
Ugh, right. To kill two birds with one stone, what you can do is, in the case of Eq, take the dual of `a = b` as `-a = -b`. The canonicalisation of this expression would give you `b - a = 0`.